File: frontend/web/src/api/products/getProduct.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getProduct = async (id: string) => {
    const token = localStorage.getItem('token');

    const { data } = await axiosInstance.get(`/api/products/${id}`, {
        headers: { authorization: `Bearer ${token}` }
    });

    return data.product;
}

export default getProduct;

File: frontend/desktop/python-client/PyQt5App/api/products/getProduct.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetProduct:
    def __init__(self, product_id):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()
        self.product_id = product_id

    def getProduct(self):
        response = requests.get(
            f"{self.address}/api/products/{self.product_id}",
            headers={
                "Authorization": f"{self.token}",
            }
        )
        data = response.json()
        print(data)
        return data.get("data", {})